### PR TITLE
Document initialConnections

### LINKED ISSUE #1317: Snaps: Document initialConnections

### PR DESCRIPTION
**Pull request by Montoya**
# Description

Updated Snaps Permissions page to include initialConnections

## Issue(s) fixed

Fixes #1317

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

**Review by FrederikBolding — APPROVED** [review]

File: docs/whats-new.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ of the [MetaMask developer page](https://metamask.io/developer/).
 
 ## May 2024
 
+- Documented [Snaps initial connections](/snaps/reference/permissions/#initial-connections).
```
Comment (alexandratran) [inline docs/whats-new.md:14]:
```suggestion
- Documented [Snaps initial connections](/snaps/reference/permissions/#initial-connections).
  ([#1318](https://github.com/MetaMask/metamask-docs/pull/1318/))
```

File: snaps/reference/permissions.md [hunk 2]
```diff
@@ -400,4 +402,35 @@ The following is an example `eth_accounts` permission:
 }
 ```
 
-The user can revoke this permission by going to the Snap's settings under **Snap permissions**.
\ No newline at end of file
+The user can revoke this permission by going to the Snap's settings under **Snap permissions**.
+
+## Initial Connections
```
Comment (alexandratran) [inline snaps/reference/permissions.md:407]:
```suggestion
## Initial connections
```

File: snaps/reference/permissions.md [hunk 3]
```diff
@@ -400,4 +402,35 @@ The following is an example `eth_accounts` permission:
...
+The following is an example of specifying `initialConnections` for a website: 
+
+```json title="snap.manifest.json"
+"initialConnections": {
+  "https://voyager-snap.linea.build": {}
+},
+```
+
+When a user visits the website and the website calls `wallet_requestSnaps`, if the Snap is already installed, the website will connect immediately and will be able to make further calls to the Snap. 
+If the Snap is not installed, the user will see a confirmation to install the Snap.
```
Comment (alexandratran) [inline snaps/reference/permissions.md:421]:
```suggestion
When a user visits the website and the website calls `wallet_requestSnaps`, if the Snap is already installed, the website connects immediately and can make further calls to the Snap. 
If the Snap is not installed, the user sees a confirmation to install the Snap. 
```

File: snaps/reference/permissions.md [hunk 4]
```diff
@@ -400,4 +402,35 @@ The following is an example `eth_accounts` permission:
...
+```json title="snap.manifest.json" 
+"initialConnections": {
+  "http://localhost:8000": {}
+},
+```
+
+It is recommended to remove local sites before deploying your Snap to production.
+
+:::note
+`initialConnections` is not a replacement for [`endowment:rpc`](#endowmentrpc). It should be used alongside the endowment to grant an automatic connection.
```
Comment (alexandratran) [inline snaps/reference/permissions.md:435]:
```suggestion
We recommend removing local sites before deploying your Snap to production.

:::note
`initialConnections` is not a replacement for [`endowment:rpc`](#endowmentrpc). Use `initialConnections` alongside `endowment:rpc` to grant an automatic connection. 
```

**Review by alexandratran — APPROVED** [review]
Minor edits, lgtm!

**Pull request merged**